### PR TITLE
devDeps: update `@metamask/eslint-config*` from 11 to 12

### PR DESCRIPTION
**Pull request by legobeat**
- #159 
- #160 
- Align peerDependency requirements
- lint fix/ignore/disable.

### CODE REVIEW

**Comment by legobeat** [comment]
@SocketSecurity ignore eslint-plugin-n@15.7.0

fs access ok

**Comment by legobeat** [comment]
@SocketSecurity ignore builtins@5.0.1

maintainership ok

File: .eslintrc.js [hunk 1]
```diff
@@ -7,6 +7,9 @@ module.exports = {
     {
       files: ['*.ts'],
       extends: ['@metamask/eslint-config-typescript'],
+      rules: {
+        '@typescript-eslint/consistent-type-imports': 'off',
```
Comment (Mrtenz) [inline .eslintrc.js:11]: Can we solve this, rather than disabling the rule?

Comment (legobeat) [reply .eslintrc.js:11]: I do believe so, but I consider that separate from the package upgrade.

Comment (legobeat) [reply .eslintrc.js:11]: But feel free to add on a commit if you think it fits better here.

Comment (mcmire) [reply .eslintrc.js:11]: Hmm. I'm having second thoughts about always deferring lint violation fixes when we upgrade lint packages by disabling ESLint rules like this. The thing is that whether we make these fixes now or later, we still have to make them, and it's easy to forget about them in the swath of other maintenance tasks we need to make time for. I understand that in other packages (e.g. `core`), we chose to disable various ESLint rules in order to lighten the load of the resulting diff, and so the compromise we made was acceptable. But I wonder if it in smaller libraries like this it wouldn't add too much to the diff to address violations for this rule at the same time as the upgrade?

**Comment by legobeat** [comment]
This was squashed, resubmitted and merged as duplicate #169

**Pull request closed without merging**